internal/db/schema/migrations/oss/postgres/88/01_worker_auth_invariant_trigger.up.sql

```diff
@@ -0,0 +1,42 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+-- update_worker_auth_authorized is a before update trigger function for the
+-- worker_auth_authorized table. The worker_auth_authorized table is a child
+-- table of server_worker. A row contains a set encryption keys for a
+-- server_worker that are unique to that worker. A server_worker can only have
+-- two rows in the worker_auth_authorized table: one with a state of 'current'
+-- and one with the state of 'previous'.
+--
+--- This trigger function ensures that there is only ever one entry
+--  with a state of 'current' and one with a state of 'previous'.
+create function update_worker_auth_authorized() returns trigger
+as $$
+begin
+    if new.state = 'current' then
+        perform
+        from worker_auth_authorized
+          where state = 'current' and worker_id = new.worker_id and worker_key_identifier != new.worker_key_identifier;
```

Comment:
This statement is a bit hard to read for me, do you mind adding a comment to elaborate on what we're checking? It looks like we can't update a row to 'current' if there is already a row with a state of 'current' with the same worker ID but a different key identifier? Isn't that always going to be true? Judging by the tests the key identifier changes every rotation.

Reply:
Er, and it won't always be true with the changes in this PR that detect when we have split brain creds and remove an incorrect `current` and update a `previous` to `current`